lib/cloud_controller/permissions.rb

```diff
@@ -233,8 +233,8 @@ def can_read_route?(space_id)
     space = VCAP::CloudController::Space.where(id: space_id).first
 
     space.has_member?(@user) || space.has_supporter?(@user) ||
-      @user.managed_organizations.map(&:id).include?(space.organization_id) ||
-      @user.audited_organizations.map(&:id).include?(space.organization_id)
+      @user.managed_organizations_dataset.where(id: space.organization_id).any? ||
```

Comment:
Same as my first comment - the `OrganizationManager` has `user_id` and `organization_id` columns, so if you use that you can cut organizations out entirely.